include/extractor/extractor.hpp

```diff
@@ -56,14 +56,18 @@ class Extractor
   private:
     ExtractorConfig config;
 
+    std::vector<TurnRestriction> ParseOSMData(ScriptingEnvironment &scripting_environment,
+                                              const unsigned number_of_threads);
```

Comment:
Looks like a separate parameter is used here because at the call site extractor takes the min of a `recommended_num_threads` and the config provided num of threads. 

Do we want to change this?